client/main.lua

```diff
@@ -54,11 +54,34 @@ end
 local function toggleRadio(toggle)
     radioMenu = toggle
     SetNuiFocus(radioMenu, radioMenu)
+
     if radioMenu then
-        exports.scully_emotemenu:playEmoteByCommand('wt')
+        local prop = lib.callback.await('qbx_radio:server:spawnProp', false)
+        local radio = NetworkGetEntityFromNetworkId(prop)
+
+        if DoesEntityExist(radio) then
+            local bone = GetPedBoneIndex(cache.ped, 28422)
+
+            local hasControl = lib.waitFor(function()
+                local isOwner = NetworkGetEntityOwner(radio) == cache.playerId
+
+                if isOwner then return true end
+
+                NetworkRequestControlOfEntity(radio)
+            end, locale('failed_get_control'), 3000)
```

Comment:
```suggestion
            NetworkRequestControlOfEntity(radio)
            local hasControl = lib.waitFor(function()
                local isOwner = NetworkGetEntityOwner(radio) == cache.playerId
                if isOwner then return true end
            end, locale('failed_get_control'), 3000)
```
Should we not request control first and wait later?

Reply:
It's checking to see if the entity owner is us first and then if it isn't it requests control.

Reply:
And it will request control every tick until it gets it. Can that degrade performance in any way?

Reply:
Probably a very very negligible amount if any but I'll adjust it in the case that I'm wrong